pgmpy/base/DAG.py

```diff
@@ -1257,6 +1257,82 @@ def copy(self):
         dag.add_nodes_from(self.nodes())
         return dag
 
+    def edge_strength(self, data, edges=None):
+        """
+        Computes the strength of edges in the DAG using conditional independence tests.
+
+        Parameters
+        ----------
+        data: pandas.DataFrame
+            The dataset on which to compute the edge strengths.
+
+        edges: tuple or list or None (default: None)
+            If None, computes strength for all edges in the DAG.
+            If tuple (X, Y), computes strength for the single edge X->Y.
+            If list [(X1, Y1), (X2, Y2), ...], computes strength for specified edges.
+
+        Returns
+        -------
+        dict
+            A dictionary mapping edges to their strength values. Each strength value
+            is based on the effect size measure of the ci_pillai test.
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> from pgmpy.base import DAG
+        >>> # Create a simple DAG
+        >>> dag = DAG([('X', 'Y'), ('Z', 'Y')])
+        >>> # Generate some data
+        >>> data = pd.DataFrame({'X': [0, 1, 0, 1], 'Y': [1, 3, 0, 2], 'Z': [1, 1, 0, 0]})
+        >>> # Compute edge strengths
+        >>> dag.edge_strength(data)
+        {('X', 'Y'): 1.2260921400386593e-07, ('Z', 'Y'): 5.140313027451882e-07}
+        >>> # Compute strength for a specific edge
+        >>> dag.edge_strength(data, edges=('X', 'Y'))
+        {('X', 'Y'): 1.2260921400386593e-07}
+        """
+        from pgmpy.estimators.CITests import pillai_trace
+
+        # If edges is None, compute for all edges in the DAG
+        if edges is None:
+            edges_to_compute = list(self.edges())
+        # If edges is a single edge tuple
+        elif isinstance(edges, tuple) and len(edges) == 2:
+            edges_to_compute = [edges]
+        # If edges is a list of edge tuples
+        else:
+            edges_to_compute = edges
+
+        strengths = {}
+
+        for edge in edges_to_compute:
+            x, y = edge
+
+            # Check if either x or y is a latent node
+            if x in self.latents or y in self.latents:
+                raise ValueError(
+                    f"Edge {edge} involves latent variables. Use CausalInference class for "
+                    "advanced causal effect estimation."
+                )
+
+            # Get parents of x and y
+            pa_X = list(self.predecessors(x))
```

Comment:
Use the `DAG.get_parents` method here. The definition of predecessors in case of causal inference and graph theory are different. In causal inference predecessor means: parents + parents of parents + their parents ... whereas in standard graph theory predecessors are defined as just the parents. 

The get_parents method makes a call to the predecessors as well, but using `get_parents` makes it more explicit here.